File: Linked List/06 Linked List Cycle.java
Link: https://leetcode.com/problems/linked-list-cycle/description/


bool hasCycle(ListNode *head) {
       ListNode slow=head;
    ListNode fast = head;
    while (fast != null && fast.next != null) {
        slow = slow.next;
        fast = fast.next.next;
        if (slow == fast) {
            return true;  // If slow and fast pointers meet, there is a cycle.
        }
    }
       return false; 
    }
}

File: Linked List/06 Linked List Cycle.cpp
Link: https://leetcode.com/problems/linked-list-cycle/description/


bool hasCycle(ListNode *head) {
        ListNode* slow=head;
        ListNode* fast=head;
        while(fast!=NULL&&fast->next!=NULL){
            slow=slow->next;
            fast=fast->next->next;
            if(slow==fast){
                return true;    // If slow and fast pointers meet, there is a cycle.
            }
        }
            return false;
    }
